client.go

```diff
@@ -0,0 +1,17 @@
+package outline
+
+import "github.com/rsjethani/secret/v2"
+
+// Client is per server top level client which acts as entry point and stores common configuration (like base url) for
+// resource level clients. It is preferred to reuse same client while communicating to the same server as this makes
+// better utilization of resources.
+type Client struct{}
```

Comment:
Do we want to introduce a sub-package for all this client/api related communication? 🤔

Reply:
Do you mean where we put actual HTTP communication code? If yes then yes that logic will not be part of this type.

Reply:
Clarified in call.
All fine now ✔️